engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/integer/AddNode.java

```diff
@@ -55,6 +62,25 @@ Object doBigIntegerLong(EnsoBigInteger self, long that) {
     return self.asDouble() + that;
   }
 
+  @Specialization(guards = "isForeignNumber(iop, that)")
+  Object doInterop(
+      Object self,
+      TruffleObject that,
+      @CachedLibrary(limit = "3") InteropLibrary iop,
+      @Cached AddNode delegate) {
+    try {
```

Comment:
I will have to add this code into all arithmetic nodes, so @Akirathan please review if [dad0fb1](https://github.com/enso-org/enso/pull/7796/commits/dad0fb1348fae7659c86298ef9aee136d5d4711b) is OK.